<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Data Preparation Guide</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Data Preparation Guide</h1>



<p>The package was developed to support the most common data shapes and
file formats used in bioinformatics. This guide describes how to
structure each file used by the portal.</p>
<p><strong>Expression matrix</strong></p>
<p>The expected format for a matrix is sample identifiers in columns and
symbols in rows. The optimal file format for the portal is an R object
of matrix type saved as an .rds file, because it will load faster,
although it is also possible to save the matrix as a CSV or TSV file. If
using the latter, ensure that the first row (with exception of the first
column) contain the sample identifiers and that the first column
contains the transcript identifiers.</p>
<blockquote>
<p><strong>Saving an RDS file</strong>: to export a matrix as an rds
file, simply run:<br />
<code>saveRDS(matrix_object, &quot;matrix_object.rds&quot;)</code></p>
</blockquote>
<p>The following is a valid example of an expression matrix:</p>
<table>
<colgroup>
<col width="5%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">S1_01</th>
<th align="right">S1_02</th>
<th align="right">S2_01</th>
<th align="right">S2_02</th>
<th align="right">S3_01</th>
<th align="right">S3_02</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ABC</td>
<td align="right">2.0376590</td>
<td align="right">-0.1768367</td>
<td align="right">0.8253947</td>
<td align="right">-0.0705087</td>
<td align="right">0.6926885</td>
<td align="right">-1.2121502</td>
</tr>
<tr class="even">
<td align="left">BCD</td>
<td align="right">0.3239560</td>
<td align="right">-1.5068365</td>
<td align="right">-1.2093268</td>
<td align="right">0.8129618</td>
<td align="right">-1.4672367</td>
<td align="right">0.9092987</td>
</tr>
<tr class="odd">
<td align="left">CDE</td>
<td align="right">1.2046363</td>
<td align="right">-0.3626300</td>
<td align="right">-0.1275494</td>
<td align="right">-2.3511435</td>
<td align="right">-0.9656587</td>
<td align="right">0.9908292</td>
</tr>
<tr class="even">
<td align="left">DEF</td>
<td align="right">2.4009177</td>
<td align="right">-1.0866808</td>
<td align="right">0.6950054</td>
<td align="right">0.7699526</td>
<td align="right">0.1906538</td>
<td align="right">1.7070437</td>
</tr>
<tr class="odd">
<td align="left">EFG</td>
<td align="right">-2.1464461</td>
<td align="right">1.0376751</td>
<td align="right">-1.2693569</td>
<td align="right">-0.7006440</td>
<td align="right">1.0708847</td>
<td align="right">2.0515956</td>
</tr>
<tr class="even">
<td align="left">FGH</td>
<td align="right">-1.0192418</td>
<td align="right">0.4986830</td>
<td align="right">1.1139692</td>
<td align="right">-1.2737894</td>
<td align="right">0.2450935</td>
<td align="right">-0.3905544</td>
</tr>
<tr class="odd">
<td align="left">GHI</td>
<td align="right">0.3026560</td>
<td align="right">0.1924001</td>
<td align="right">-0.6198241</td>
<td align="right">-0.5189834</td>
<td align="right">0.8785241</td>
<td align="right">0.4883541</td>
</tr>
<tr class="even">
<td align="left">HIJ</td>
<td align="right">-1.7404140</td>
<td align="right">0.3375572</td>
<td align="right">1.0659999</td>
<td align="right">-0.3770236</td>
<td align="right">-0.8395515</td>
<td align="right">-0.4042909</td>
</tr>
<tr class="odd">
<td align="left">IJK</td>
<td align="right">0.3681634</td>
<td align="right">-1.1588354</td>
<td align="right">0.2511265</td>
<td align="right">0.1662243</td>
<td align="right">1.1440126</td>
<td align="right">0.7103129</td>
</tr>
<tr class="even">
<td align="left">JKL</td>
<td align="right">0.5101581</td>
<td align="right">-1.2043604</td>
<td align="right">-1.8345244</td>
<td align="right">-0.0596510</td>
<td align="right">-0.0566549</td>
<td align="right">0.8097529</td>
</tr>
</tbody>
</table>
<p><strong>Measures table</strong></p>
<p>The measures table follows the format of one row per subject (even if
they have more than one sample collected) with the measures across
columns. A data.frame can be saved in an .rds file, but CSV or TSV files
are also supported.</p>
<p>Measures collected over time should be represented in separate
columns, with the convention (enforced by default) of a time code as a
suffix for measure names, separated by underscore (<code>_</code>) –
this means that underscore cannot be used in long measure names as well.
For example, for disease activity collected over four time points, the
expected names are: diseaseActivity_Baseline, diseaseActivity_Week1,
diseaseActivity_Week2 and diseaseActivity_Week3. Using the default
settings, it is invalid to use a name such as Disease_Activity_Baseline.
The time separator can be modified in the configuration file by setting
<code>timesep</code> to the desired separator.</p>
<p>The following is a valid example of a measures table:</p>
<table>
<thead>
<tr class="header">
<th align="left">Patient_ID</th>
<th align="right">Platelets_m01</th>
<th align="right">Platelets_m02</th>
<th align="right">Age</th>
<th align="left">drugNaive</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">p01</td>
<td align="right">182.6504</td>
<td align="right">160.8753</td>
<td align="right">44</td>
<td align="left">Yes</td>
</tr>
<tr class="even">
<td align="left">p02</td>
<td align="right">178.4293</td>
<td align="right">231.0919</td>
<td align="right">70</td>
<td align="left">Yes</td>
</tr>
<tr class="odd">
<td align="left">p03</td>
<td align="right">217.7025</td>
<td align="right">191.2578</td>
<td align="right">75</td>
<td align="left">No</td>
</tr>
</tbody>
</table>
<p><strong>Lookup table</strong></p>
<p>For datasets where a subject has more than one sample (e.g. samples
over time, from different tissues or combinations thereof), a lookup
table should be constructed and saved as a data.frame in an .rds file,
CSV or TSV.</p>
<p>This table maps subject identifiers to sample identifiers, with the
expression matrix containing data for <em>all</em> samples in the
dataset. The table should also contain metadata that allows subsetting
samples. For example, if subjects and samples vary over time, drug
groups and tissues, the lookup table should have one column for each
category. The following is an example of such a table:</p>
<table>
<thead>
<tr class="header">
<th align="left">Sample_ID</th>
<th align="left">Time</th>
<th align="left">Tissue</th>
<th align="left">Drug</th>
<th align="left">Patient_ID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">S1_01</td>
<td align="left">m01</td>
<td align="left">A</td>
<td align="left">d1</td>
<td align="left">p01</td>
</tr>
<tr class="even">
<td align="left">S1_02</td>
<td align="left">m02</td>
<td align="left">A</td>
<td align="left">d1</td>
<td align="left">p01</td>
</tr>
<tr class="odd">
<td align="left">S2_01</td>
<td align="left">m01</td>
<td align="left">A</td>
<td align="left">d1</td>
<td align="left">p02</td>
</tr>
<tr class="even">
<td align="left">S2_02</td>
<td align="left">m02</td>
<td align="left">A</td>
<td align="left">d1</td>
<td align="left">p02</td>
</tr>
<tr class="odd">
<td align="left">S3_01</td>
<td align="left">m01</td>
<td align="left">A</td>
<td align="left">d2</td>
<td align="left">p03</td>
</tr>
<tr class="even">
<td align="left">S3_02</td>
<td align="left">m02</td>
<td align="left">A</td>
<td align="left">d2</td>
<td align="left">p03</td>
</tr>
</tbody>
</table>
<p>In the case above, patients p01 and p02 belong to drug group d1,
while patient p03 belongs to group d2. All patients have samples
collected at months 1 and 2 (encoded as m01 and m02), and all samples
are from the same tissue (A). In this example, drug groups samples from
different subjects (i.e. there is no overlap of subjects between the two
drug groups).</p>
<p>The lookup table can also be enriched with other characteristics of
subjects that can be use to partition samples, such as age, sex, or
others. Outputs of methods such as clustering can also be added to the
table: this enables the exploration of correlations in different
clusters, or comparing trajectories across different clusters, for
example.</p>
<div id="validation-checks" class="section level3">
<h3>Validation checks</h3>
<p>The package does a very lightweight validation of the loaded files,
only checking if subjects and samples match. It does not ensure that the
correct transformations have been applied to the expression data, nor
does it warn about or modify missing data – a subject is not included in
calculations if they have a missing value for a particular measure.</p>
<p>The following checks ARE made:</p>
<p><strong>Matching samples and subjects</strong>: the package will
confirm that every sample in the expression matrix is matched to at
least one subject in the lookup table. It also checks that all subjects
in the measures table match to at least one sample in the lookup table.
That is, there can be no excess of samples or subjects in each
table.</p>
<p><strong>Matrix format:</strong> if using an .rds file, the package
will check that the expression matrix was indeed saved as a matrix
object in the rds file. This is to ensure that the rownames are read
properly.</p>
</div>
<div id="additional-files" class="section level2">
<h2>Additional files</h2>
<div id="differential-expression-analysis-results" class="section level3">
<h3>Differential expression analysis results</h3>
<p>The package includes two modules to showcase results of differential
expression analysis (see <a href="config.html">config</a> for more
details). These modules read files created using <code>limma</code>,
<code>edgeR</code> or <code>deseq2</code>. All files should be saved
with column names and the column names must not be changed – the only
exception is you want to mix models from different packages, then you
should rename the columns so that all results have the same column names
(e.g. p-values are identified in the same way across all files).</p>
<p>These modules require the creation of a table that lists all model
results and they support the use of additional columns in the table to
organize results from different types of models or subsets of samples.
All model results file should be placed into a <code>models</code>
folder within the project folder.</p>
<p>The table should look like the following and saved in a CSV or TSV
file:</p>
<table>
<thead>
<tr class="header">
<th align="left">Model</th>
<th align="left">Time</th>
<th align="left">Drug</th>
<th align="left">File</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Linear</td>
<td align="left">m01</td>
<td align="left">d1</td>
<td align="left">Model_1.txt</td>
</tr>
<tr class="even">
<td align="left">Linear</td>
<td align="left">m02</td>
<td align="left">d2</td>
<td align="left">Model_2.txt</td>
</tr>
<tr class="odd">
<td align="left">Nonlinear</td>
<td align="left">m01</td>
<td align="left">d1</td>
<td align="left">Model_3.txt</td>
</tr>
<tr class="even">
<td align="left">Nonlinear</td>
<td align="left">m02</td>
<td align="left">d2</td>
<td align="left">Model_4.txt</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-moduleslists" class="section level3">
<h3>Gene modules/lists</h3>
<p>The heatmap module requires the creation of a table containing lists
of names such as gene symbols (see <a href="config.html">config</a> for
more details). In this table, each row will have a column that contains
the gene lists, with symbols separated by a comma. If you have a table
where you have a list identifier and a symbol in each column, you can
use a group-by operation with paste-collapse to create the required
list, as follows:</p>
<div style="display: flex;">
<div>
<p>Original file:</p>
<table>
<thead>
<tr class="header">
<th align="left">module</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">ABC</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="left">DEF</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">GHI</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">JKL</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="left">MNO</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">PQR</td>
</tr>
</tbody>
</table>
</div>
<div class="col">
<p>  <!-- an empty Div (with a white space), serving as
a column separator --></p>
</div>
<div>
<p>Code to transform:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>table <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">transmute</span>(<span class="at">list =</span> <span class="fu">paste</span>(gene, <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>))</span></code></pre></div>
</div>
</div>
</div>
<div id="asking-for-help" class="section level3">
<h3>Asking for help</h3>
<p>If you have any issues with data preparation, please post it as an
issue on the package GitHub.</p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
